### PR TITLE
test(tagging): add coverage for SemanticEntry, PdfTag Clone/Debug, and heading-level clamping

### PR DESCRIPTION
**Pull request by mitsuru**
## Summary

`crates/fulgur/src/tagging.rs` のリージョンカバレッジを **93.72% → 95.09%**（+1.37 pp）に改善しました。

このファイルは最もカバレッジが低いファイルでしたが、その主な原因は以下の 2 点でした：

- `SemanticEntry` 構造体が既存テストで一度も構築されておらず、`#[derive(Debug, Clone)]` で生成されたコードが未カバー
- `PdfTag` の `Debug` 実装（全 16 バリアント）が未呼び出し（`assert_eq!` は `PartialEq` のみ使用し、テストが全部 pass するため `Debug` は呼ばれない）

## Changes

- `SemanticEntry` の構築テスト（`parent` / `alt_text` の Some/None 全組み合わせ）
- `SemanticEntry::clone` の動作確認テスト（フィールドが正しく複製されること）
- `SemanticEntry::Debug` のフォーマットテスト（`format!("{:?}", entry)` を初めて呼び出し）
- `PdfTag::Clone` の全バリアントテスト（unit バリアント 14 種 + フィールド付き 3 種）
- `PdfTag::Debug` のフォーマットテスト（全 16 バリアント）
- `pdf_tag_to_krilla_tag` の heading level clamping edge case テスト（level=0 → H1、level=7/255 → H6）
- `classify_element` の追加 edge case（HTML `<figure>` → `None`、`<a>`/`<link>` → `None` の意図を文書化）

**意図的にカバーしなかった箇所：**
- テストコード内の `assert!(matches!(...))` / `if let` の false 分岐（これらは tests がすべて pass する限り unreachable であり、カバーすることに意味がない）

## Test plan

- [x] `cargo test -p fulgur --lib`（1904 passed）
- [x] `cargo clippy -- -D warnings`（警告なし）
- [x] `cargo fmt --check`（差分なし）
- [ ] `npx markdownlint-cli2 '**/*.md'`（ドキュメント変更なし）

---
_Generated by [Claude Code](https://claude.ai/code/session_01MKQqczGB2sDbKDmLB9qaUZ)_

## Summary by CodeRabbit

* **テスト**
  * `SemanticEntry` の構築時に `parent`／`alt_text` の None/Some が正しく反映されることを検証しました。
  * `SemanticEntry` の `Clone` が全フィールドを保持し、`Debug` 出力に `SemanticEntry` とタグ名が含まれることを確認しました。
  * `PdfTag` の `Clone`（フィールド保持/ユニット）の同一性と、`Debug` 表現の期待文字列を検証しました。
  * 見出しレベルの範囲外入力（0・上限超過）は内部でクランプされても `Hn` 相当になることを確認しました。
  * `classify_element` で `<figure>`、`<a>`、`<link>` が `None` を返すことを追加検証しました。

### CODE REVIEW

File: crates/fulgur/src/tagging.rs [hunk 1]
```diff
@@ -369,4 +369,193 @@ mod tests {
...
+        assert!(matches!(k, krilla::tagging::TagKind::Hn(_)));
+    }
+
+    #[test]
+    fn pdf_tag_to_krilla_tag_heading_level_above_max_clamped_to_six() {
+        // level=7 and level=255 are above the PDF maximum H6; both clamp to 6.
+        let k7 = pdf_tag_to_krilla_tag(&PdfTag::H { level: 7 }, None, None);
+        let k255 = pdf_tag_to_krilla_tag(&PdfTag::H { level: 255 }, None, None);
+        assert!(matches!(k7, krilla::tagging::TagKind::Hn(_)));
+        assert!(matches!(k255, krilla::tagging::TagKind::Hn(_)));
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/tagging.rs]: ご指摘ありがとうございます。`6a81f02` で修正しました。`TagKind::Hn(tag)` をデストラクチャして `tag.level().get()` を呼び出すことで、クランプ後の実際の見出しレベル値（`level=0 → 1`、`level=7/255 → 6`）まで検証するようにしました。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/tagging.rs [hunk 2]
```diff
@@ -369,4 +369,193 @@ mod tests {
...
+    fn semantic_entry_debug_contains_struct_name_and_tag() {
+        let entry = SemanticEntry {
+            tag: PdfTag::Table,
+            parent: Some(3),
+            alt_text: None,
+        };
+        let s = format!("{entry:?}");
+        assert!(s.contains("SemanticEntry"));
+        assert!(s.contains("Table"));
+    }
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/tagging.rs:461]: ご指摘ありがとうございます。`6a81f02` で修正しました。`semantic_entry_debug_contains_struct_name_and_tag` に以下の assert を追加し、フィールド値まで検証するようにしました。

- `s.contains("Some(3)")` — `parent` フィールドの値
- `s.contains("None")` — `alt_text: None`
- `PdfTag::L { numbering: ListNumbering::Disc }` の Debug 出力に `"Disc"` を含むこと
- `PdfTag::Th { scope: TableHeaderScope::Column }` の Debug 出力に `"Column"` を含むこと

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/tagging.rs [hunk 3]
```diff
@@ -369,4 +369,193 @@ mod tests {
...
+
+    #[test]
+    fn pdf_tag_to_krilla_tag_heading_level_above_max_clamped_to_six() {
+        // level=7 and level=255 are above the PDF maximum H6; both clamp to 6.
+        let k7 = pdf_tag_to_krilla_tag(&PdfTag::H { level: 7 }, None, None);
+        let k255 = pdf_tag_to_krilla_tag(&PdfTag::H { level: 255 }, None, None);
+        assert!(matches!(k7, krilla::tagging::TagKind::Hn(_)));
+        assert!(matches!(k255, krilla::tagging::TagKind::Hn(_)));
+    }
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/tagging.rs:578]: ご指摘ありがとうございます。`6a81f02` で修正しました。`TagKind::Hn(tag)` をデストラクチャして `tag.level().get()` で `NonZeroU16` の値を直接 assert するようにしました（`level=0 → 1`、`level=7/255 → 6`）。バリアント一致だけでは `clamp` ロジックの検証が不十分という点、ご指摘のとおりです。

---
_Generated by [Claude Code](https://claude.ai/code)_
